addons/http_routing/models/ir_http.py

```diff
@@ -488,6 +488,11 @@ def _dispatch(cls):
                     return redirect
                 elif url_lg:
                     request.uid = None
+                    if request.httprequest.path == '/%s/' % url_lg:
+                        # special case for homepage controller
```

Comment:
more detailed comment + comment it is same as post_process_args + missing if querystring before append